src/main/java/org/mule/module/apikit/validation/attributes/HeadersValidator.java

```diff
@@ -109,30 +116,44 @@ private void validateHeadersStrictly(Map<String, String> headers) throws Invalid
 
   private void validateHeader(List<String> values, String name, Parameter type)
       throws InvalidHeaderException {
-    if (values.isEmpty())
+    if (values.isEmpty()) {
       return;
-
-    if (values.size() > 1 && !type.isArray() && !type.isRepeat())
+    }
+    if (values.size() > 1 && !type.isArray() && !type.isRepeat()) {
       throw new InvalidHeaderException("Header " + name + " is not repeatable");
-
-    // raml 1.0 array validation
+    }
     if (type.isArray()) {
-      validateType(name, values, type);
+      validateTypeValues(name, explodeCommaSeparatedValues(values), type);
     } else {
-      // single header or repeat
-      validateType(name, values.get(0), type);
+      validateTypeValue(name, values.get(0), type);
     }
   }
 
-  private void validateType(String name, List<String> values, Parameter type) throws InvalidHeaderException {
-    final StringBuilder yamlValue = new StringBuilder();
-    for (String value : values)
-      yamlValue.append("- ").append(value).append("\n");
+  private List<String> explodeCommaSeparatedValues(List<String> listOfCsv) {
+    try {
+      return listOfCsv.stream().map(v -> "[" + v + "]")
+          .map(JSONArray::new)
+          .collect(ArrayList::new, (list, jsonArray) -> jsonArray.forEach(v -> list.add(v.toString())),
+                   ArrayList::addAll);
+    } catch (JSONException e) {
+      //If there is some issue parsing the value as array continue with the actual value
+    }
+    return explodeCommaSeparatedScalarValues(listOfCsv);
+  }
```

Comment:
shouldn't we assume that if the header is repeated the values are already separated ?

Reply:
Great question! I was expecting it to be asked because I assumed by my own that array headers eventually could be sent multiple times as CSV lists to support anything. We can discuss this and make the assumption that we think makes more sense

Reply:
The use case that I think could bring some trouble with this approach is when the header is defined as string[] and send ["Hi, how are you", "fine, thanks"] -> ["Hi" , "how are you", "fine", "thanks"]

Reply:
This is not an issue as soon as the value is double quoted. The parsing of `["Hi, how are you", "fine, thanks"]` in this case will end up with 2 values `Hi, how are you` and `fine, thanks`.
I added these test cases for this use case:
- success-with-including-quoted-string-array-header
- success-with-csv-including-quoted-string-array-header

Reply:
But the current behaviour if the values are not quoted would change, right?

Reply:
Right. If someone is sending a String array with a single value containing a comma, APIKit is considering it as a single String value, while making this change will cause APIKit to consider it as two separated header values.
Anyway, as per the RFC, commas are delimiters
 https://tools.ietf.org/html/rfc7230#section-3.2.6
```
   Most HTTP header field values are defined using common syntax
   components (token, quoted-string, and comment) separated by
   whitespace or specific delimiting characters.  Delimiters are chosen
   from the set of US-ASCII visual characters not allowed in a token
   (DQUOTE and "(),/:;<=>?@[\]{}").
```
and if you want to include a comma as part of the value, you should send it as a `quoted-string`
```
   A string of text is parsed as a single value if it is quoted using
   double-quote marks.

     quoted-string  = DQUOTE *( qdtext / quoted-pair ) DQUOTE
     qdtext         = HTAB / SP /%x21 / %x23-5B / %x5D-7E / obs-text
     obs-text       = %x80-FF
```